From 8d31326bf15dd3f6eaaddd61c450ade9f3979f7b Mon Sep 17 00:00:00 2001
From: sub77 <sub77@ymail.com>
Date: Mon, 7 Mar 2016 09:22:18 -0600
Subject: [PATCH 2/3] Add new setting TW_DEVICE_SPECIFIC_VERSION

* If TW_DEVICE_SPECIFIC_VERSION is set in BoardConfig, it will
* override the TW_VERSION_STR which is set in recovery/variables.h

Change-Id: I195b62bcc11976a4073d85c96bdeaacd38c9bb04
Signed-off-by: sub77 <sub77@ymail.com>
---
 Android.mk  | 3 +++
 variables.h | 4 ++++
 2 files changed, 7 insertions(+)

diff --git a/Android.mk b/Android.mk
index 62ef11a..ed1a882 100644
--- a/Android.mk
+++ b/Android.mk
@@ -341,6 +341,9 @@ ifneq ($(TW_DEFAULT_LANGUAGE),)
 else
     LOCAL_CFLAGS += -DTW_DEFAULT_LANGUAGE=en
 endif
+ifneq ($(TW_DEVICE_SPECIFIC_VERSION),)
+    LOCAL_CFLAGS += -DTW_DEVICE_SPECIFIC_VERSION=\"$(TW_DEVICE_SPECIFIC_VERSION)\"
+endif
 
 LOCAL_ADDITIONAL_DEPENDENCIES := \
     dump_image \
diff --git a/variables.h b/variables.h
index 241c7dc..63e6969 100644
--- a/variables.h
+++ b/variables.h
@@ -17,7 +17,11 @@
 #ifndef _VARIABLES_HEADER_
 #define _VARIABLES_HEADER_
 
+#ifdef TW_DEVICE_SPECIFIC_VERSION
+#define TW_VERSION_STR TW_DEVICE_SPECIFIC_VERSION
+#else
 #define TW_VERSION_STR              "3.0.2-0"
+#endif
 
 #define TW_USE_COMPRESSION_VAR      "tw_use_compression"
 #define TW_FILENAME                 "tw_filename"
-- 
2.7.4

